torch/fx/experimental/symbolic_shapes.py

```diff
@@ -269,6 +269,12 @@ def eval(cls, base, divisor):
             if isinstance(base, FloorDiv):
                 return FloorDiv(base.args[0], base.args[1] * divisor)
 
+            if isinstance(base, sympy.Add):
+                for a in base.args:
+                    gcd = sympy.gcd(a, divisor)
+                    if gcd == divisor:
+                        return FloorDiv(base - a, divisor) + a / gcd
```

Comment:
The logic here is a bit inefficient. It would be better to do a single pass over `base` rather than recursively constructing a new one for each term. Also you don't need to do a full gcd if all you are checking is if gcd == divisor. You can just see if the one divides the other, e.g., using `div`. Note that `sympy.gcd` does *polynomial* gcd. Mathematically, that is fine for what you are doing here, but if you intended for this to be integer gcd, that's `igcd` (and [currently doesn't support symbolic arguments](https://github.com/sympy/sympy/issues/8530)).

Reply:
Also below, you can avoid `simplify` and use `div` instead. `simplify` is usually a bad idea for library code because it can have bad performance if you give it a complex expression, since it tries *everything* to try to simplify things. If you know the exact simplification function you need, it's always better to use that instead.

Reply:
Or more to the point, you can just use `cancel`.

Reply:
igcd won't work here, these are symbolic